src/Api.Worker/MessageConsumers/DepositUpdatedConsumer.cs

```
@@ -127,7 +127,25 @@ public async Task Consume(ConsumeContext<DepositUpdated> context)
                 BrokerAccountName = brokerAccount.Name
             };
 
-            await _depositRepository.Upsert(deposit);
+            DepositUpdate depositUpdate = null;
+            try
+            {
+                depositUpdate = await _depositRepository.Upsert(deposit);
+            }
+            catch (InvalidOperationException)
+            {
+                depositUpdate = await _depositRepository.GetUpdateOrDefault(evt.DepositId, evt.Sequence);
+
+                if (depositUpdate == null)
+                    throw;
```

It's better to throw a new exception here